server/src/com/thoughtworks/go/server/websocket/ConsoleLogSender.java

```diff
@@ -88,11 +88,21 @@ public void process(final SocketEndpoint webSocket, JobIdentifier jobIdentifier,
                 }
             } while (webSocket.isOpen() && !detectCompleted(jobIdentifier));
 
+            LOGGER.debug("Sent {} log lines for {} from {}", streamer.totalLinesConsumed(), jobIdentifier, consoleService.consoleLogFile(jobIdentifier).toPath());
             // empty the tail end of the file because the build could have been marked completed, and exited the
             // loop before we've seen the last content update
             if (isRunningBuild) sendLogs(webSocket, streamer, jobIdentifier);
 
-            LOGGER.debug("Sent {} log lines for {}", streamer.totalLinesConsumed(), jobIdentifier);
+            //send the remaining logs if any
+            if (detectCompleted(jobIdentifier)) {
+                ConsoleConsumer consoleFileStreamer = consoleService.getStreamer(start, jobIdentifier);
+                start += sendLogs(webSocket, consoleFileStreamer, jobIdentifier);
+                LOGGER.debug("Sent {} log lines for {} from {}", streamer.totalLinesConsumed(), jobIdentifier, consoleService.consoleLogFile(jobIdentifier).toPath());
+                consoleFileStreamer.close();
```

Comment:
You probably mean to do this in a finally block (or try-with-resources), should something blow up between opening the file and closing it.

Otherwise, squash and merge...